### PR TITLE
Add require section inside exports

### PR DESCRIPTION
**Pull request by shrujalshah28**
Partially Fix: #177

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,6 +13,11 @@
           "types": "./index.d.ts",
           "default": "./dist/index.esm.js"
         },
+        "require": {
+          "types": "./index.d.ts",
```
Comment (12wrigja) [inline package.json:17]: I tried repro-ing this locally but couldn't get compilation to succeed when this was a .d.ts file. It did work when it was a .d.cts file, but that would mean duplicating the types file we maintain, and I'd prefer if we didn't have to do that (or, if we only did it as part of a build process so we can have a single source of truth).

Comment (shrujalshah28) [reply package.json:17]: With this, I know we still have an error but after this change, we can get intelligence support in IDE.

Comment (shrujalshah28) [reply package.json:17]: Do you think we can merge this?
I am open to submitting a follow-up PR to address your suggestion? or Do you think I just submit that change in this PR?

Comment (shrujalshah28) [reply package.json:17]: https://github.com/js-temporal/temporal-polyfill/pull/184

Comment (12wrigja) [reply package.json:17]: Sorry for the delayed responses - I was on vacation.

Do I understand correctly that with just this change you were seeing type information in an editor when you weren't before, even though compilation shouldn't be succeeding? That sounds like a bug to me.

Do you have a minimal reproduction + instructions that I can follow?

Comment (12wrigja) [reply package.json:17]: I'm not opposed at all to finding and merging a fix for this issue, I just want to be sure that it's the right combination of fixes.

**Comment by 12wrigja** [comment]
Closing this out as we instead merged https://github.com/js-temporal/temporal-polyfill/pull/184.

**Pull request closed without merging**